crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -975,7 +977,7 @@ impl CumulocityConverter {
             }
             _ => {
                 // if device is unregistered register using auto-registration
-                if self.entity_store.get(&source).is_none() {
+                if self.entity_store.get(&source).is_none() && self.config.enable_auto_register {
```

Comment:
I'd move that `self.config.enable_auto_register` check inside this `if` block and provide an `else` case that will return an error that" the provided entity: {entity_topic_id} was not found and could not be auto-registered either, because it is disabled".